src/gwc/src/main/java/org/geoserver/gwc/layer/DefaultTileLayerCatalog.java

```diff
@@ -138,8 +140,12 @@ public DefaultTileLayerCatalog(
     private XStream newXStream() {
         XStream serializer = this.xstreamProvider.get();
         serializer.allowTypeHierarchy(GeoServerTileLayerInfo.class);
+        serializer.allowTypes(new Class[] {DimensionWarning.WarningType.class});
         // have to use a string here because UnmodifiableSet is private
         serializer.allowTypes(new String[] {"java.util.Collections$UnmodifiableSet"});
+        serializer.allowTypes(new Class[] {DimensionWarning.WarningType.class});
```

Comment:
I think that line 143 and 146 are identical

Reply:
Yep, fixing